embedded/remotestorage/s3/s3.go

```diff
@@ -691,4 +715,127 @@ func (s *Storage) scanObjectNames(ctx context.Context, prefix string, limit int)
 	return entries, subPaths, nil
 }
 
+func (s *Storage) getRoleCredentials() error {
+	if !s.S3RoleEnabled {
+		return nil
+	}
+
+	if s.accessKeyID != "" || s.secretKey != "" {
```

Comment:
this validation could even be done before constructing the main remote storage struct, e.g. Validate method in options

Reply:
I did not find the `Validate` method for the regular `Options` or `RemoteStorageOptions`. Should I create one?
Additionally, this is the check that would only work in that niche case. Check beforehand might be a bit of a help but not by much.

Reply:
it could at least be early validated at `createRemoteStorageInstance` for instance. It may be simpler to validate options upfront and in one place. Also raising the error right from the start before giving any chance to do any further work that could end-up in an error being thrown while using it.

Ideally there could be a validate options method, we already have that level of uniformity under `embedded` package, but not in the server, at least not yet. But it does not need to be done right now. Having the validations upfront and in a single place would be enough